glib/gsocket.go

```diff
@@ -0,0 +1,28 @@
+package glib
+
+import "C"
+import (
+	"errors"
+	"syscall"
+	"unsafe"
+)
+
+func SocketNew(domain, typ, proto int) (*Object, error) {
```

Comment:
Please do not return `*Object` here, but rather create a new `Socket` struct that wraps `*Object`. This also makes it possible to implement the `ValueTransformer` interface.

Depending on your use case an unmarshaler may also be needed.

Reply:
done